app/src/main/java/com/example/plannr/admin/adminEdit/AdminEditPresenter.java

```diff
@@ -0,0 +1,345 @@
+
+        //retrieve input info
+        String courseCode = view.getEditCourseCode().replaceAll("\\s", "").toUpperCase(Locale.ROOT);
+        String courseName = view.getEditCourseName();
+        boolean fall = view.getEditFallAvailability();
+        boolean winter = view.getEditWinterAvailability();
+        boolean summer = view.getEditSummerAvailability();
+        String prerequisites = view.getEditPrerequisite().replaceAll("\\s", "").toUpperCase(Locale.ROOT);
+
+        //check if valid
+        if(inputValidator(courseCode, courseName, fall, winter, summer, prerequisites) == true){
+
+            //get the given courses id
+            readData(new FirebaseCallback() {
+                @Override
+                public void onCallBack(HashMap<String, String> list) {
+
+                    //convert input prerequisites into list
+                    Course course1 = new Course();
+                    ArrayList<String> givenPrerequisites = course1.stringToArraylist(prerequisites);
+                    TextView warningText = view.getEditWarningText();
+
+                    //insert the choice of no prerequisite
+                    list.put("", "");
+
+                    //check if prerequisites exist in database
+                    int count = 0;
+
+                    for(int i = 0; i < givenPrerequisites.size(); i++){
+                        if(list.containsValue(givenPrerequisites.get(i))){
+                            count ++;
+                        }
+                    }
+
+                    if(count == givenPrerequisites.size()){
+
+
+                        getCourse(new FirebaseCallback2() {
+                            @Override
+                            public void findSelection(String code) {
+                                //Hide warning message
+                                warningText.setTextColor(Color.GREEN);
+                                warningText.setText("THE COURSE WAS UPDATED!");
+
+                                //create prerequisite id
+                                String idPrerequisites = "";
+
+                                for(int i = 0; i < givenPrerequisites.size(); i++){
+                                    for(Map.Entry<String, String> set : list.entrySet()){
+                                        if(set.getValue().equals(givenPrerequisites.get(i)) && givenPrerequisites.get(i).equals("") == false){
+                                            idPrerequisites =  idPrerequisites + "," + set.getKey();
+                                        }
+                                    }
+                                }
+
+                                Log.i("PREREQUISITE", idPrerequisites);
+
+
+                                //get given course id
+                                String id = "";
+
+                                for(Map.Entry<String, String> set: list.entrySet()){
+                                    if(set.getValue().equals(code)){
```

Comment:
what does `set.getValue()` return? in line 95 it seems to be used for comparing the prerequisite courses, but here it seems to be used to compare to the course code?

Reply:
The map is composed of ids as keys and course codes as the value. On line 95, we are comparing our prerequisite course codes to the maps/db course codes, and if they match we want to take their id, which is the key of the map at the index where they match. In line 107 we follow the same logic, but now we only want to convert the selected course code for editing into its id form.

Reply:
Cool! Thanks for explaining